PythonTool/TagExpansion.py

```diff
@@ -0,0 +1,177 @@
+                        )
+
+    args = parser.parse_args()
+
+    return args
+
+
+def get_vector_tags(mb, element):
+    """
+    Create a list of each vector tag in the mesh file by identifying each tag on
+    a specific mesh element and determining if it is vector or scalar.
+
+    Input:
+    ______
+       mb: Core
+           A PyMOAB core instance with a loaded data file.
+       element: int
+           The type of MOAB element from which to extract the tag list.
+
+    Returns:
+    ________
+       element_list: List
+           The list of all specific elements in the mesh.
+       vector_tags: List
+           A list of all vector tags in the mesh.
+    """
+
+    # Retrieve an arbitrary MBHEX element in the mesh and extract the tag list.
+    root = mb.get_root_set()
+    element_list = mb.get_entities_by_type(root, element)
+
+    # If there are none of the specified mesh elements, print a warning and exit.
+    if len(element_list) == 0:
+        print("WARNING: No hex elements were found in this data file.")
+        exit()
+
+    tag_list = mb.tag_get_tags_on_entity(element_list[0])
+
+    # Check if each tag is a vector. If so, append it to the list.
+    vector_tags = []
+    for tag in tag_list:
+        tag_length = mb.tag_get_length(tag)
+        if tag_length > 1:
+            vector_tags.append(tag)
+
+    return element_list, vector_tags
+
+
+def tag_expansion(mesh_file, output_file = None):
+    """
+    Expand the vector tags on each element in the given mesh data file. Write a
+    file to disk for each index with the corresponding scalar tag values.
+
+    Input:
+    ______
+       mesh_file: str
+           User supplied mesh file location.
+       output_file: str
+           Optional user supplied output file name.
+
+    Returns:
+    ________
+       none
+    """
+
+    # Load the mesh file to create a reference mesh.
+    mb_ref = core.Core()
+    mb_ref.load_file(mesh_file)
+
+    # Retrieve the list of vector tags on the mesh.
+    hexes_ref, vec_tags_ref = get_vector_tags(mb_ref, types.MBHEX)
+
+    # Retrieve the length and name of the vector tag.
+    reference_length = vec_tags_ref[0].get_length()
+    tag_name = vec_tags_ref[0].get_name()
+
+    # Delete each vector tag from the reference mesh.
+    for tag in vec_tags_ref:
+        mb_ref.tag_delete(tag)
+
+    # Load the mesh file for extracting vector tag data.
+    mb_ext = core.Core()
+    mb_ext.load_file(mesh_file)
+
+    # Retrieve the list of vector tags on the mesh.
+    hexes_ext, vec_tags_ext = get_vector_tags(mb_ext, types.MBHEX)
+
+    # Make sure the mesh file contains at least one vector tag.
+    if len(vec_tags_ext) < 1:
+        print("WARNING: This mesh file did not contain any vector tags.")
+        exit()
+
+    # Warn the user if the mesh file contains more than one type of vector tag.
+    if len(vec_tags_ext) < 1:
+        print("WARNING: This mesh file contains elements with more than one vector tag.")
```

Comment:
does this script need to exit if more than one is found? Or will it continue working?

Reply:
It should exit, since a huge refactor is needed to handle multiple vector tags!